task/blue-green-deploy/0.1/blue-green-deploy.yaml

```diff
@@ -0,0 +1,75 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: blue-green-deploy
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: deployment
+    tekton.dev/displayName: "blue green deployment"
+spec:
+  description: >-
+    This task can be used to do Blue-Green deployment
+  workspaces:
+    - name: manifest-dir
+      description: Consisting of kubernetes manifests
+    - name: kubeconfig-dir
+      description: Can be used in case kubeconfig file is mounted
+  params:
+    - name: SERVICE_NAME
+      type: string
+      description: Name of the service which is pointing to existing deployment
+    - name: NEW_VERSION
+      type: string
+      description: The version of newer deployment which is to be patched to existing service
+    - name: MANIFEST
+      type: string
+      default: "."
+      description: The newer version of the deployment to be deployed in another zone
+  steps:
+    - name: change-deployment
+      image: quay.io/openshift/origin-cli:latest
+      workingDir: $(workspaces.manifest-dir.path)
+      script: |
+        #!/usr/bin/env bash
+
+        if [[ -f $(workspaces.kubeconfig-dir.path)/kubeconfig ]]; then
+          export KUBECONFIG=$(workspaces.kubeconfig-dir.path)/kubeconfig
+        fi
+
+        getReadyStatus() {
+          local READY
+          i=0
+          for f in $(kubectl get deploy $1 -o jsonpath='{range .status.conditions[*]}{.reason}{"\n"}{end}')
+          do
+            if [[ $f == "MinimumReplicasAvailable" ]]; then
+              READY=$(kubectl get deploy $1 -o jsonpath='{range .status.conditions['$i']}{.status}{"\n"}{end}')
+              if [[ "$READY" != "True" ]]; then
+                return 1
+              else
+                return 0
+              fi
+            fi
+            ((i++))
+          done
+          return 1
```

Comment:
it seems `return 1` is used to indicate `true` but the convention is to return 0 to indicate `true` e.g. `foo || bar` reads if `foo` fails then run `bar`. fails meaning returns non zero